src/test/scala/modbat/mbt/ModbatTestHarness.scala

```diff
@@ -71,45 +96,36 @@ object ModbatTestHarness {
     current_comparison
   }
 
-  def filtering_and_regex(name_output : String, typeoutput : String, typeByte : ByteArrayOutputStream, name_file: String){
-    var regex_list = List("")
+  def filtering_writing_comparing_each_file(name_output_1 : String, name_output_2 : String, typeByte : ByteArrayOutputStream){
+    var regex_map = Map[String, String]()
```

Comment:
The long name suggests that it is better to break up this function into two parts.
Use "camel case" notation, also in the functions before, e.g., "compareIterators".

Reply:
Also, make sure that the validated template that is *read* has extension .out or .eout, but the files that are *written* with the current output end in .log or .err, respectively.